streams-runtimes/streams-runtime-threaded/src/main/java/org/apache/streams/threaded/controller/ThreadingController.java

```diff
@@ -65,15 +66,37 @@
      * The threading controller
      */
     public static ThreadingController getInstanceLowPriority() {
-        return INSTANCE_LOW_PRIORITY;
+        synchronized (ThreadingController.class) {
```

Comment:
FYI, this is the same as putting "synchronized" in the method signature since the method is static.  Same as synchronizing on "this" for an instance method.